cmake/dts.cmake

```diff
@@ -49,6 +49,11 @@ if(SUPPORTS_DTS)
       dts_files
       ${DTC_OVERLAY_FILE_AS_LIST}
       )
+    # Allow overlays to include dtsi from the board directory
+    list(APPEND
+      DTS_ROOT_SYSTEM_INCLUDE_DIRS
+      -isystem ${BOARD_DIR}
+      )
```

Comment:
Let us instead add the BOARD_DIR to the list of DTS_ROOT's.

Then it will be easier to document this behaviour, and we will need less CMake code.

Unfortunately . is not in the list of include paths for DTS_ROOTs, so any .dtsi file that needs to be included from an application overlay would need to first be first moved to BOARD_DIR/include, or BOARD_DIR/dts. But this is a small price to pay for system consistency IMHO.

Reply:
This makes sense, but it's applying a standard to a new case: there are no board definitions that have `dts` or `include` subdirectories, but there are board definitions that have `dtsi` files in `.`.  This is probably the first time board-specific DTS options have been exercised.

Based on previous experience with #12377, which has some slight relation to this, I'd like confirmation/acknowledgement from @galak and @nashif that this approach is acceptable before moving things around.

Reply:
Changes made.